receiver/scraperhelper/scrapercontroller.go

```diff
@@ -78,6 +78,11 @@ func NewScraperControllerReceiver(
 		return nil, errors.New("collection_interval must be a positive duration")
 	}
 
+	timeout := cfg.Timeout
+	if timeout == 0 {
+		timeout = cfg.CollectionInterval
+	}
```

Comment:
Docs say the timeout *"must be within the range of (0, CollectionInterval]"*. I think instead of doing this we should enforce this requirement and return an error if the timeout is 0. You can already set the timeout to the CollectionInterval if you want to.

Reply:
So this point was raised in a previous PR that was merged that @astencel-sumo highlighted is that this change (not yet made it as part of a release) will enforce users to set both `timeout` and `collection_interval` if they wanted to reduce the `collection_interval` below the default `timeout`.

This allows the zero value (actual 0) to represent, "Use collection interval as timeout", which was the previous behaviour to this change. The expectation will be that `timeout` fields will only be used for scrapers with large collection intervals that doesn't make sense to persist the connection for 10m or 30m.